Tasks/VsTestPlatformToolInstallerV1/nugetfeedinstaller.ts

```diff
@@ -103,9 +126,52 @@ export class NugetFeedInstaller {
         }
 
         vstestPlatformInstalledLocation = await new NugetDownloadHelper(this.consolidatedCiData)
-            .attemptPackageDownload(packageSource, testPlatformVersion);
+            .attemptPackageDownload(packageSource, testPlatformVersion, nugetConfigFilePath);
 
         // Set the vstest platform tool location for the vstest task to consume
         helpers.setVsTestToolLocation(vstestPlatformInstalledLocation);
     }
+
+    // Utility function that writes the feed url along with username and password if provided into the specified nuget config file
+    private prepareNugetConfigFile(packageSource: string, configFilePath: string, username: string, password: string) : string {
+        const feedUrl = tl.getInput(constants.customFeed);
+        const feedId = uuid.v1();
+
+        tl.debug(`Writing package source details to temp config file ${configFilePath}`);
+
+        try {
+            // Write the skeleton nuget config contents to the config file
+            fs.writeFileSync(configFilePath, constants.emptyNugetConfig, { encoding: 'utf-8' });
+        } catch (error) {
+            this.consolidatedCiData.failureReason = 'configFileWriteFailed';
+            throw new Error(tl.loc('ConfigFileWriteFailed', configFilePath, error));
+        }
+
+        if (!helpers.isNullEmptyOrUndefined(password) && helpers.isNullEmptyOrUndefined(username)) {
+            username = 'vstestPlatformToolInstaller';
+        }
+
+        const nugetTool = tl.tool(path.join(__dirname, 'nuget.exe'));
+
+        nugetTool.arg(constants.sources).arg(constants.add).arg(constants.noninteractive)
+            .arg(constants.name).arg(feedId).arg(constants.source).arg(feedUrl)
+            .argIf(password, constants.usernameParam).argIf(password, username)
+            .argIf(password, constants.passwordParam).argIf(password, password)
+            .argIf(configFilePath, constants.configFile).argIf(configFilePath, configFilePath);
+
+        this.consolidatedCiData.prepareConfigFileStartTime = perf();
+        const result = nugetTool.execSync();
+        this.consolidatedCiData.prepareConfigFileEndTime = perf();
+
+        if (result.code !== 0 || !(result.stderr === null || result.stderr === undefined || result.stderr === '')) {
+            this.consolidatedCiData.failureReason = constants.configFileWriteFailed;
+            throw new Error(tl.loc('ConfigFileWriteFailed', configFilePath, result.stderr));
+        }
+
+        // Assign the feed name we wrote into the config file to the packageSource variable
+        tl.debug(`Setting the source to feed with id ${feedId} whose details were written to the config file.`);
+        ci.publishEvent('PackageSourceOverridden', {packageSource: 'customFeed'} );
+
+        return feedId;
```

Comment:
this is doing two things 
1. generating nuget config file
2. getting feed id 

please split it.

Reply:
yep. Found the need for this while writing UTs. Will do this

Reply:
done